### PR TITLE
Gas price minimum basefee opcode

### PR DESCRIPTION
**Pull request by gastonponti**
The blockchain team is changing the way we are going to store the gasPriceMinimum.
Following that ethereum did, we are going to store the gasPriceMinimum for Celo, as a field in the header called `baseFee`. We will also restore the baseFee opcode.
The client it's not going to update the gasPriceMinimum anymore. Maintaining this contract after the fork, consuming the baseFee instead, is a way to be backwards compatible for the contracts from the community that use it. The client will stop to use it.

The problem now, is that the basefee operation is introduced in the solidity version 0.8.7, and our repo is way behind that.
An alternative could be to call a Wrapper contract that only returns the `block.basefee` field but that contract will need to be in the version 0.8.7 which again, is incompatible with what we have in our repo

### CODE REVIEW

File: packages/protocol/contracts/common/GasPriceMinimum.sol [hunk 1]
```diff
@@ -60,19 +64,22 @@ contract GasPriceMinimum is
    * @param _gasPriceMinimumFloor The lowest value the gas price minimum can be.
    * @param _targetDensity The target gas fullness of blocks, expressed as a fixidity fraction.
    * @param _adjustmentSpeed How quickly the minimum changes, expressed as a fixidity fraction.
+   * @param _baseFeeOpCodeActivationBlock Block number where the baseFee opCode is activated
```
Comment (montera82) [inline packages/protocol/contracts/common/GasPriceMinimum.sol]: nit 
```suggestion
   * @param _baseFeeOpCodeActivationBlock Block number where the baseFee opCode is activated.
```

File: packages/protocol/contracts/common/GasPriceMinimum.sol [hunk 2]
```diff
@@ -108,25 +115,45 @@ contract GasPriceMinimum is
     emit GasPriceMinimumFloorSet(_gasPriceMinimumFloor);
   }
 
+  /**
+   * @notice Set the activation block of the baseFee opCode.
+   * @param _baseFeeOpCodeActivationBlock Block number where the baseFee opCode is activated
```
Comment (montera82) [inline packages/protocol/contracts/common/GasPriceMinimum.sol]: nit 
```suggestion
   * @param _baseFeeOpCodeActivationBlock Block number where the baseFee opCode is activated.
```

File: packages/protocol/contracts/common/GasPriceMinimum.sol [hunk 3]
```diff
@@ -108,25 +115,45 @@ contract GasPriceMinimum is
...
+      "baseFee opCode activation block must be greater than zero"
+    );
+    baseFeeOpCodeActivationBlock = _baseFeeOpCodeActivationBlock;
+    emit BaseFeeOpCodeActivationBlockSet(_baseFeeOpCodeActivationBlock);
+  }
+
   /**
    * @notice Retrieve the current gas price minimum for a currency.
    * @param tokenAddress The currency the gas price should be in (defaults to gold).
    * @return current gas price minimum in the requested currency
```
Comment (montera82) [inline packages/protocol/contracts/common/GasPriceMinimum.sol]: nit 
```suggestion
   * @return current gas price minimum in the requested currency
```

File: packages/protocol/contracts/common/GasPriceMinimum.sol [hunk 4]
```diff
@@ -108,25 +115,45 @@ contract GasPriceMinimum is
...
    */
   function getGasPriceMinimum(address tokenAddress) external view returns (uint256) {
+    uint256 _gasPriceMinimum;
+    if (baseFeeOpCodeActivationBlock > 0 && block.number >= baseFeeOpCodeActivationBlock) {
+      _gasPriceMinimum = block.basefee;
+    } else {
+      _gasPriceMinimum = gasPriceMinimum;
+    }
```
Comment (montera82) [inline packages/protocol/contracts/common/GasPriceMinimum.sol:143]: are we planning to add a test case that covers this?

File: packages/protocol/contracts/common/GasPriceMinimum.sol [hunk 5]
```diff
@@ -108,25 +115,45 @@ contract GasPriceMinimum is
...
+   */
+  function setBaseFeeOpCodeActivationBlock(uint256 _baseFeeOpCodeActivationBlock) public onlyOwner {
+    require(
+      _baseFeeOpCodeActivationBlock > 0,
+      "baseFee opCode activation block must be greater than zero"
+    );
+    baseFeeOpCodeActivationBlock = _baseFeeOpCodeActivationBlock;
+    emit BaseFeeOpCodeActivationBlockSet(_baseFeeOpCodeActivationBlock);
+  }
+
```
Comment (montera82) [inline packages/protocol/contracts/common/GasPriceMinimum.sol:131]: are we planning to add a test for this case?

File: packages/protocol/contracts/common/GasPriceMinimum.sol (hunk 4 above)
Comment (gastonponti) [reply packages/protocol/contracts/common/GasPriceMinimum.sol:143]: Yes, I'm waiting for the 0.8.7 version resolution, of if we switch strategies

**Comment by martinvol** [comment]
Closed in favour of https://github.com/celo-org/celo-monorepo/pull/10379

**Pull request closed without merging**